source/solvers/tracer_assemblers.cc

```diff
@@ -698,3 +698,98 @@ TracerAssemblerBoundaryNitsche<dim>::assemble_rhs(
 
 template class TracerAssemblerBoundaryNitsche<2>;
 template class TracerAssemblerBoundaryNitsche<3>;
+
+template <int dim>
+void
+TracerAssemblerReaction<dim>::assemble_matrix(
+  const TracerScratchData<dim> &scratch_data,
+  StabilizedMethodsCopyData    &copy_data)
+{
+  // Scheme and physical properties
+  const std::vector<double> &k = scratch_data.tracer_reaction_constant;
+  const double               a = scratch_data.tracer_reaction_order;
+
+  // Loop and quadrature informations
+  const auto        &JxW_vec    = scratch_data.JxW;
+  const unsigned int n_q_points = scratch_data.n_q_points;
+  const unsigned int n_dofs     = scratch_data.n_dofs;
+
+  auto &strong_jacobian_vec = copy_data.strong_jacobian;
+  auto &local_matrix        = copy_data.local_matrix;
+
+  // assembling local matrix
+  for (unsigned int q = 0; q < n_q_points; ++q)
+    {
+      const double reaction_coeff =
+        k[q] * a *
+        std::pow(scratch_data.tracer_values[q], a - 1); // k * a * C^(a-1)
+
+      // Update the strong Jacobian with the reaction term contribution
+      for (unsigned int j = 0; j < n_dofs; ++j)
+        {
+          const auto phi_T_j = scratch_data.phi[q][j];
+          // Strong Jacobian contribution: - k * a * C^(a-1) phi
+          strong_jacobian_vec[q][j] += -reaction_coeff * phi_T_j;
+        }
+
+      // Store JxW in local variable for faster access;
+      const double JxW = JxW_vec[q];
+
+      for (unsigned int i = 0; i < n_dofs; ++i)
+        {
+          const auto phi_T_i = scratch_data.phi[q][i];
+
+          for (unsigned int j = 0; j < n_dofs; ++j)
+            {
+              const auto phi_T_j = scratch_data.phi[q][j];
```

Comment:
```suggestion
              const auto &phi_T_j = scratch_data.phi[q][j];
```

No need to copy these.